tests/components/statistics/test_sensor.py

```diff
@@ -557,7 +557,7 @@ async def test_age_limit_expiry(hass: HomeAssistant) -> None:
 
         # Values expire over time. Buffer is empty
 
-        current_time += timedelta(minutes=1)
+        current_time += timedelta(minutes=1, seconds=1)
```

Comment:
Why is this changed?

Reply:
This is an edge case. The old "purge" function already removed a value when it is sitting exactly at the edge of the defined interval (open vs. closed interval / "<" vs. "<="), whereas now it is being kept until it is really out. This is why I had to add one more second to make the expiring value "really out of the max age range". Actually, a microsecond would have done the job as well.

For any real-world scenario outside of unit tests this doesn't matter anyway, since this edge case will not be relevant. Real measurements will very likely either be a few microseconds before or after.

However, it will become relevant for some further changes I am planning to make (with separate PRs), that will make sure that the computation covers the full time frame (max age).  This PR was created to split up one that was covering a couple of different issues, and that is why the buffer always holds all values that are observable in the specified interval (including the last change before the interval, or in the edge case, the change at the edge of the interval).